library/rsa.c

```diff
@@ -1193,64 +1200,145 @@ int mbedtls_rsa_rsassa_pkcs1_v15_sign( mbedtls_rsa_context *ctx,
         if( mbedtls_oid_get_oid_by_md( md_alg, &oid, &oid_size ) != 0 )
```

Comment:
@andresag01 I have no strong feelings about that, but sticked to what it was before. Actually, for both alternatives there's a scenario where it would make sense - if the user specifies a valid entry of `mbedtls_md_type_t` which however is not supported, `FEATURE_UNAVAILABLE` would be appropriate, while if it's just a random integer with no meaning to `mbedtls_md_type_t`, then `BAD_INPUT_DATA` would make more sense.